content/blog/adopting-existing-cloud-resources-into-pulumi/index.md

```diff
@@ -0,0 +1,242 @@
+ =   └─ aws:ec2:Vpc       my-vpc             import     
+ 
+Resources:
+    = 1 to import
+    1 unchanged
+```
+
+Because the `import` is provided in code, it can be configured in many different ways.  We've seen users look up import ids from Pulumi config instead of hardcoding into their programs, construct them from predicatable names based on existing program parameters, or even conditionally add the `import: <id>` property depending on whether they want to deploy a stack in "adopt" mode or in "create" mode.
+
+This allows `import` to be used for a wide variety of adoption scenarios - from importing a single resource, to migrating a whole stack from an existing tool like Terraform to Pulumi, and even to automating the migration process across dozens of instances of an infrastructure deployment.
+
+### Walkthrough of Adopting Existing Infrastructure to Pulumi
+
+To see one of these scenarios in action end-to-end, let's walk through the process of deploying some "existing" infrastructure outside of Pulumi, and then adopting it under management of a Pulumi program.  You can also check out the video below which walks through the same scenario.
+
+<iframe width="560" height="315" src="https://www.youtube.com/embed/kX_3Wdft0Ms" frameborder="0" allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>
+
+We'll use Terraform to provision our "existing" infrastructure.  These resources could have been provisioned manually in the console, or using platform-native infrastructure-as-code solutions like CloudFormation, ARM or Kubernetes YAML.
+
+We'll also use TypeScript, though the process would work mostly the same in Python as well.
+
+#### 1. Deploy "existing" infrastructure
+
+For our "existing" infrastructure, we'll use a small Azure deployment of a Virtual Machine and corresponding Resource Group and Networking resources.  We'll deploy [this example](https://www.terraform.io/docs/providers/azurerm/r/virtual_machine.html#example-usage-from-an-azure-platform-image-) using `terraform`.  Save the code in that example to `main.tf` and run `terraform init` and then `terraform apply` to deploy it.
+
+#### 2. Create a Pulumi program
+
+Next, we'll create a Pulumi program to describe the same infrastructure that exists in Azure.  We could write this by hand, but since we used Terraform to define the original infrastructure, we also have the option to use [`tf2pulumi`](https://github.com/pulumi/tf2pulumi) to automatically convert our `.tf` files to Pulumi.  
+
+```bash
+$ pulumi new azure-typescript --force
+$ tf2pulumi < main.tf > index.ts
+```
+
+We now have a Pulumi program that describes the same infrastructure:
+
+```ts
+import * as pulumi from "@pulumi/pulumi";
+import * as azure from "@pulumi/azure";
+
+const config = new pulumi.Config();
+const prefix = config.get("prefix") || "tfvmex";
+
+const mainResourceGroup = new azure.core.ResourceGroup("main", {
+    location: "West US 2",
```

Comment:
add a comment somewhere about region, or suggest config?

Reply:
This is the literal code emitted by tf2pulumi and a match for the TF example that was the seed for this. We can leave as is.